### ISSUE TITLE
Channel has neither first nor second edge when running GetNetworkInfo

### ISSUE THREAD

**Issue by aakselrod** [lightningnetwork/lnd#128]
## Describe the Bug

As reported by `Ylbam` on IRC, the following panic occurred:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x68 pc=0x43a9ab]

goroutine 409 [running]:
panic(0xb2ee00, 0xc04200a090)
        /dev/go/src/runtime/panic.go:500 +0x1af
main.(*rpcServer).GetNetworkInfo.func3(0x0, 0x0, 0x10bf6c0, 0xc042416000)
        /dev/go/src/github.com/lightningnetwork/lnd/rpcserver.go:1605 +0x6b
github.com/lightningnetwork/lnd/channeldb.(*ChannelGraph).ForEachChannel.func1.1(0x3fb9890, 0x8, 0x8, 0x3fb9898, 0x42, 0x42, 0x42, 0x0)
        /dev/go/src/github.com/lightningnetwork/lnd/channeldb/graph.go:190 +0x333
github.com/lightningnetwork/lnd/vendor/github.com/boltdb/bolt.(*Bucket).ForEach(0xc0424df140, 0xc0422cf8a8, 0xa, 0xa)
        /dev/go/src/github.com/lightningnetwork/lnd/vendor/github.com/boltdb/bolt/bucket.go:361 +0x106
github.com/lightningnetwork/lnd/channeldb.(*ChannelGraph).ForEachChannel.func1(0xc0426bfa40, 0xd99680, 0xc0426bfa40)
        /dev/go/src/github.com/lightningnetwork/lnd/channeldb/graph.go:191 +0x179
github.com/lightningnetwork/lnd/vendor/github.com/boltdb/bolt.(*DB).View(0xc0424f4000, 0xc0422cf950, 0x0, 0x0)
        /dev/go/src/github.com/lightningnetwork/lnd/vendor/github.com/boltdb/bolt/db.go:626 +0xbc
github.com/lightningnetwork/lnd/channeldb.(*ChannelGraph).ForEachChannel(0xc0422cf9f8, 0xc0422cfa38, 0xc0425669a0, 0x0)
        /dev/go/src/github.com/lightningnetwork/lnd/channeldb/graph.go:192 +0x66
main.(*rpcServer).GetNetworkInfo(0xc0424b7230, 0x3d71800, 0xc0421da6c0, 0x11144c8, 0x0, 0xc0424ffac8, 0x611ec3)
        /dev/go/src/github.com/lightningnetwork/lnd/rpcserver.go:1624 +0x308
github.com/lightningnetwork/lnd/lnrpc._Lightning_GetNetworkInfo_Handler(0xbf5e00, 0xc0424b7230, 0x3d71800, 0xc0421da6c0, 0xc0422d83c0, 0x0, 0x0, 0x0, 0xc0425781d8, 0x1)
        /dev/go/src/github.com/lightningnetwork/lnd/lnrpc/rpc.pb.go:3454 +0x284
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc0421fa000, 0x10cc680, 0xc042446680, 0xc042000800, 0xc042572390, 0x10ec8e0, 0xc04225e150, 0x0, 0x0)
        /dev/go/src/github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/server.go:708 +0xbdf
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc.(*Server).handleStream(0xc0421fa000, 0x10cc680, 0xc042446680, 0xc042000800, 0xc04225e150)
        /dev/go/src/github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/server.go:892 +0x6ab
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0425661b0, 0xc0421fa000, 0x10cc680, 0xc042446680, 0xc042000800)
        /dev/go/src/github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/server.go:468 +0xb2
created by github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
        /dev/go/src/github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/server.go:469 +0xaa
```

## Steps to Reproduce

Unknown at this time. Additional information may be forthcoming from `Ylbam`.

## Investigation so Far

This appears to be an issue in a switch inside a closure inside the `GetNetworkInfo` RPC handler. The channel has neither the first nor second edge populated, and the switch doesn't handle the case where both are `nil`.

## Next Steps

Preliminarily, we should add a check for both edges being `nil` inside the switch. Secondarily, we should try and trace what code might add non-conforming edges to the DB.

**Comment by aakselrod** [comment]
`Ylbam` has reported another panic with a similar cause, this time in `func (r *ChannelRouter) syncChannelGraph(syncReq *syncRequest) error` in `rouing/router.go`. Panic is listed below:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x70c284]

goroutine 71 [running]:
panic(0xb2ee80, 0xc04200a090)
        C:/dev/go/src/runtime/panic.go:500 +0x1af
github.com/lightningnetwork/lnd/routing.(*ChannelRouter).syncChannelGraph.func2(0x0, 0x0, 0x10bf6c0, 0xc04242e1a0)
        D:/dev/go/src/github.com/lightningnetwork/lnd/routing/router.go:835 +0xa44
github.com/lightningnetwork/lnd/channeldb.(*ChannelGraph).ForEachChannel.func1.1(0x3dfa728, 0x8, 0x8, 0x3dfa730, 0x42, 0x42, 0x42, 0x0)
        D:/dev/go/src/github.com/lightningnetwork/lnd/channeldb/graph.go:190 +0x333
github.com/lightningnetwork/lnd/vendor/github.com/boltdb/bolt.(*Bucket).ForEach(0xc042243c80, 0xc0426f59a8, 0xa, 0xa)
        D:/dev/go/src/github.com/lightningnetwork/lnd/vendor/github.com/boltdb/bolt/bucket.go:361 +0x106
github.com/lightningnetwork/lnd/channeldb.(*ChannelGraph).ForEachChannel.func1(0xc0425f8fc0, 0xd99720, 0xc0425f8fc0)
        D:/dev/go/src/github.com/lightningnetwork/lnd/channeldb/graph.go:191 +0x179
github.com/lightningnetwork/lnd/vendor/github.com/boltdb/bolt.(*DB).View(0xc042174000, 0xc0426f5a50, 0x0, 0x0)
        D:/dev/go/src/github.com/lightningnetwork/lnd/vendor/github.com/boltdb/bolt/db.go:626 +0xbc
github.com/lightningnetwork/lnd/channeldb.(*ChannelGraph).ForEachChannel(0xc042132018, 0xc0426f5b20, 0x0, 0x0)
        D:/dev/go/src/github.com/lightningnetwork/lnd/channeldb/graph.go:192 +0x66
github.com/lightningnetwork/lnd/routing.(*ChannelRouter).syncChannelGraph(0xc042486750, 0xc042138010, 0x23, 0xc042245e30)
        D:/dev/go/src/github.com/lightningnetwork/lnd/routing/router.go:903 +0x157
github.com/lightningnetwork/lnd/routing.(*ChannelRouter).networkHandler(0xc042486750)
        D:/dev/go/src/github.com/lightningnetwork/lnd/routing/router.go:518 +0x1727
created by github.com/lightningnetwork/lnd/routing.(*ChannelRouter).Start
        D:/dev/go/src/github.com/lightningnetwork/lnd/routing/router.go:216 +0x181
```

**Comment by aakselrod** [comment]
Fixed by commit 5560f032f62abe82364eaa6e1b5d79234472da1e
